File: cmstestsuite/code/oom-static.c
#include <stdio.h>

int big[125 * 1000 * 1000 / sizeof(int)];

int main() {
    int i;
    // If we don't do this cycle, the compiler is smart enough not to
    // map the array into resident memory.
    for (i = 0; i < 125 * 1000 * 1000 / sizeof(int); i++) {
      big[i] = 0;
    }
    scanf("%d", &big[10000]);
    printf("correct %d\n", big[10000]);
    return 0;
}

File: cmstestsuite/code/oom-static.pas
program correct;

var
   big: array[0..125 * 1000 * 1000 div sizeof(integer)] of integer;
   i: longint;

begin
     for i := 1 to 125 * 1000 * 1000 div sizeof(integer) do
       big[i] := 0;
    readln(big[10000]);
    writeln('correct ', big[10000]);
end.
